### PR TITLE
fix NPE taglib notfound w skipmetainf

### LINKED ISSUE #20247: webContainer property skipMetaInfResourcesProcessing=true can cause NullPointerException in JSP taglib
**Describe the bug**  
In Liberty, if webContainer property skipMetaInfResourcesProcessing is true, and a JSP references a taglib with a path under WEB-INF, it can cause a NullPointerException on the call to the JSP resulting in just the message Error 500: java.lang.NullPointerException.   The error in messages.log is as follows:

java.lang.NullPointerException
        at java.base/java.io.File.<init>(File.java:278)
        at com.ibm.ws.jsp.webcontainerext.JspDependent.isOutdated(JspDependent.java:84)
        at com.ibm.ws.jsp.webcontainerext.JspDependent.toString(JspDependent.java:129)
        at java.base/java.text.MessageFormat.subformat(MessageFormat.java:1302)
        at java.base/java.text.MessageFormat.format(MessageFormat.java:885)
        at java.base/java.text.Format.format(Format.java:158)
        at java.base/java.text.MessageFormat.format(MessageFormat.java:860)
        at com.ibm.ws.logging.internal.impl.BaseTraceFormatter.formatMessage(BaseTraceFormatter.java:301)
        at com.ibm.ws.logging.internal.impl.BaseTraceFormatter.formatVerboseMessage(BaseTraceFormatter.java:373)
        at com.ibm.ws.logging.internal.impl.BaseTraceService.publishTraceLogRecord(BaseTraceService.java:1072)
        at com.ibm.ws.logging.internal.impl.BaseTraceService.publishLogRecord(BaseTraceService.java:1043)
        at com.ibm.ws.logging.internal.impl.BaseTraceService$2.publish(BaseTraceService.java:285)
        at java.logging/java.util.logging.Logger.log(Logger.java:979)
        at com.ibm.ws.logging.internal.WsLogger.log(WsLogger.java:138)
        at com.ibm.ws.logging.internal.WsLogger.logp(WsLogger.java:336)
        at com.ibm.ws.logging.internal.WsLogger.exiting(WsLogger.java:465)
        at com.ibm.ws.jsp.webcontainerext.JspDependent.<init>(JspDependent.java:58)
        at com.ibm.ws.jsp.webcontainerext.AbstractJSPExtensionServletWrapper.loadClassInformation(AbstractJSPExtensionServletWrapper.java:233)
        at com.ibm.ws.jsp.webcontainerext.AbstractJSPExtensionServletWrapper.handleRequest(AbstractJSPExtensionServletWrapper.java:214)
        at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1227)
        at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5049)
        ... 20 more
 

**Steps to Reproduce**  
See above.

**Expected behavior**  
The JSP should return to the client without a NPE even in file not found situations.

### PR DESCRIPTION
**Pull request by epj**
fixes #20247 
An NPE can result if a taglib refers to a file where the path is under WEB-INF and the webContainer property skipMetaInfResourcesProcessing="true".

### CODE REVIEW

File: dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspSkipMetaInfServer/bootstrap.properties [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+###############################################################################
+bootstrap.include=../testports.properties
+osgi.console=7777
+com.ibm.ws.logging.trace.specification=*=info:com.ibm.ws.webcontainer*=all:com.ibm.wsspi.webcontainer*=all:HTTPChannel=all:TCPChannel=all:GenericBNF=all:com.ibm.ws.jsp*=all
+
```
Comment (volosied) [inline dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspSkipMetaInfServer/bootstrap.properties]: Could you address this new line?

File: dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPSkipMetaInfTests.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright (c) 2013, 2021 IBM Corporation and others.
```
Comment (volosied) [inline dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPSkipMetaInfTests.java]: Copyright 2022

Comment (epj) [reply dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPSkipMetaInfTests.java]: done

**Comment by epj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jsp_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.jsp_fat_lWAS,com.ibm.ws.jsp_fat_jsp.2.3.toleration,com.ibm.ws.el.3.0_fat

**Comment by epj** [comment]
No test failures.   Build had red status due to child build not responding.

**Review by volosied — APPROVED** [review]
Everything looks good. Thanks

**Comment by epj** [comment]
#libbybot

**Pull request merged**